### PR TITLE
Ivan Golubev

### PR DESCRIPTION
**Pull request by rjkward**

### CODE REVIEW

**Comment by akorolczuk** [comment]
Pretty basic stuff, would say "maybe" for mid level, "no" for senior

File: ivan-golubev/TankBatallion/Assets/Scripts/Managers/ScoreManager.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        }
+
+        public int GetHighScore() { return scores[0]; }
+        public int[] GetScores() { return scores; }
+
+        public void AddNewScore(int newScore)
+        {
+            if (scores[scores.Length - 1] > newScore)
+                return;
+            for (int i = scores.Length - 1; i >= 0; i--)
```
Comment (Moniraego) [inline ivan-golubev/TankBatallion/Assets/Scripts/Managers/ScoreManager.cs]: I don't know if he just always wants to right high perf code or if he doesn't know or trust C# libraries but he could have just kept this array as a list and added it then called a sort function. Subsequently he has a bug here I guess: the new score replaces an existing score but others aren't shifted down to push one of them out.

File: ivan-golubev/TankBatallion/Assets/Scripts/ObjectPools/WallEffectPool.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace TankBatallion
+{
+    public class WallEffectPool : MonoBehaviour
```
Comment (Moniraego) [inline ivan-golubev/TankBatallion/Assets/Scripts/ObjectPools/WallEffectPool.cs]: I would imagine someone trying to impress us would have these pools as one generic pool type.

**Comment by Moniraego** [comment]
@akorolczuk Considering that this is a big assignment, I don't think 'basic' is a very good criticism. I think these should weed out people who have problematic implementations. Do you see any problem with this implementation, other than it being pretty minimalistic?

**Comment by akorolczuk** [comment]
@Moniraego ok I can be more specific: 
- the only object oriented pattern he used is a sigleton and it's incorrectly implemented 
- everything including logic/data is a MonoBehaviour. There are weird dependencies, like GameManager having reference to HUD view 
- Creates a pool for each object instead of using generic one.

This just look like junior/mid level. The candidate I'd like to hire would cascade his knowledge to other members of the team, mentor them, come up with ideas to improve various parts of the game. I just don't see that capability here. 

If you want you can conduct a technical interview with him but to me it looks like a waste of time.

**Pull request closed without merging**